Docs Update: Minor Update to the order of concepts and renames

karthikscale3 opened the pull request:
Another incremental minor update for the sake of better clarity. Shown below the pre post of the changes. Happy to revert back anything you feel is unnecessary. Also, these changes DO NOT affect the permalinks so it won't cause any breakages to existing links on the internet.

![image](https://github.com/user-attachments/assets/eec2681b-31c9-4997-bbc3-9fae842d58cb)

krypticmouse commented:
It looks good for the most part but this might break google index in the short term, so I suggest to add some redirection links in mkdocs

Check #1681 for reference

krypticmouse commented on docs/mkdocs.yml:
```diff
@@ -72,15 +77,10 @@ nav:
-      - BFRS: deep-dive/optimizers/bfrs.md
-      - CoPro: deep-dive/optimizers/copro.md
-      - MIProV2: deep-dive/optimizers/miprov2.md
+    - Data Handling:
```
Can we push data handling to the top again please?

karthikscale3 replied:
done

karthikscale3 commented:
> It looks good for the most part but this might break google index in the short term, so I suggest to add some redirection links in mkdocs
> 
> Check #1681 for reference
> 
> Thank you so much for the contribution!!

Good call. This PR does not actually change the links. I verified it and it keeps the old links. I just added one redirect for the assertions doc inside deep-dive section which was duplicated so I removed it and added a redirect for it.

krypticmouse commented:
Awesome! LGTM :ship: 

Thank you so much for the contribution!!
